bin/ih-setup

```diff
@@ -174,23 +174,38 @@ function ih::private::apply-to-steps() {
     done
   fi
 
+  local RESULT=0, FAILED_STEP
+
+
   for STEP in "${STEPS[@]}"; do
     # This will be filtering the steps down to only those which are in domain
     DOMAIN=${STEP%.*}
     # We want to look for the step with spaces on each side, which is annoying to quote
     # shellcheck disable=SC2027
     if [[ " ${DOMAINS[*]} " =~ " "${DOMAIN}" " ]]; then
       if [[ " ${IH_SETUP_STEPS[*]} " =~ " "${STEP}" " ]]; then
-        if ! "ih::private::$COMMAND" "$STEP"; then
-          ih::log::error "Command $COMMAND failed for step $STEP. Re-run using 'ih-setup -v install' for more details."
-          exit 1
+        "ih::private::$COMMAND" "$STEP"
+        RESULT=$?
+        RESULT=$RESULT || 0
```

Comment:
These lines seem unnecessary? Can't you check `if [[ $? -gt 0 ]]` directly?

Reply:
updated the code to use this instead and added a flag if there's an error